File: chapters/chapter2/scripts/matlab/solveFEM.m
function u = solveFEM(bc,A0,F0)
  u_s0 =[]; u_s1=[];
  if bc(1,1)==1 , u_s0=bc(1,3) ; end
  if bc(2,1)==1 , u_s1=bc(2,3) ; end
  u0=A0\F0;
  u=[u_s0;u0;u_s1];
end

File: chapters/chapter2/scripts/python/solveFEM.py
def solveFEM(bc,A0,F0):
    u_s0 =[]; u_s1 =[] 
    if bc[0,0]==1: u_s0=bc[0,2] 
    if bc[1,0]==1: u_s1=bc[1,2]
    u0=sparse.linalg.spsolve(A0,F0)
    u=u_s0
    u=numpy.append(u,u0)
    u=numpy.append(u,u_s1)
    return u